cetlvast/suites/docs/examples/example_05_variable_length_array_vs_vector.cpp

```diff
@@ -0,0 +1,181 @@
+/// @file
+/// Demonstration of the differences between std::vector and cetl::variable_length_array type.
+///
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+///
+
+#include "cetl/pmr/array_memory_resource.hpp"
+#include "cetl/pf17/cetlpf.hpp"
+#include "cetl/variable_length_array.hpp"
+#include <iostream>
+#include <vector>
+
+template <typename T>
+void add_hello_world(T& container)
+{
+    container.reserve(12);
+    container.push_back('H');
+    container.push_back('e');
+    container.push_back('l');
+    container.push_back('l');
+    container.push_back('o');
+    container.push_back(' ');
+    container.push_back('W');
+    container.push_back('o');
+    container.push_back('r');
+    container.push_back('l');
+    container.push_back('d');
+}
+
+template <typename T>
+void print_container(const T& container)
+{
+    for (auto c : container)
+    {
+        std::cout << c;
+    }
+
+    std::cout << std::endl;
+}
+
+int main()
+{
+#if !__cpp_exceptions
+    {
+        //! [example_no_exceptions]
+
+        // Using the cetlpf.hpp header we can create a polymorphic allocator that is aliased to
+        // std::pmr::polymorphic_allocator when compiling with C++17 or newer and cetl::pf17::pmr::polymorphic_allocator
+        // when compiling with C++14.
+        cetl::pmr::polymorphic_allocator<int> alloc{cetl::pmr::new_delete_resource()};
```

Comment:
Bug:  Template type mismatch among alloc (int) and a/b (char).  Why does this compile?

Reply:
everything was signed and no implicit casts narrowed any types so, while it is wrong (and I fixed it), it would only waste memory (but might be more efficient for some memory resources on some platforms due to alignment).